### ISSUE TITLE
Weird behaviour when delete a profile of a urlmap

### ISSUE THREAD

**Issue by GabrielLasso** [curiefense/curiefense#544]
**Describe the bug**
When I click to delete a profile, I don't know what happens, but it goes wrong (and I become unable to save, like in https://github.com/curiefense/curiefense/issues/543)
**To Reproduce**
1. Go to urlmaps
2. Fork some profiles
3. Delete a profile
4. Now the profiles are wrong

**Expected behavior**
Delete the selected profile

**Screenshots**
![Gravar_select-area_20210806114818](https://user-images.githubusercontent.com/12103372/128529044-a2409e4d-6289-4924-81a3-fc008154df0c.gif)

**Desktop (please complete the following information):**
 - OS: Manjaro
 - Browser: Opera

**Additional context**
Running on main. I deployed it yesterday

**Comment by jdorfman** [comment]
Thanks @GabrielLasso for the detailed report and gif 🙌

I will send this over to @Aviv-Galmidi

**Comment by spiros-p** [comment]
I just encountered the same bug. It seems that deleting a forked profile will _appear_ to assign its Match value back to an existing profile (overwriting whatever it had before), but the assignment isn't actually made.

Steps to illustrate this:
- After deleting the forked profile, do not save anything. 
- If more than one URL Map is defined, go to the pulldown on the upper right and switch the editor to a different URL Map, then return to the original one. The incorrect Match value is still displayed, which makes it appear as if the assignment was saved somehow.
- However, if you switch the editor to a different document type (e.g., Rate Limits), and then back to URL Maps, the incorrect value is no longer displayed.

**Comment by GabrielLasso** [comment]
@spiros-p 
> However, if you switch the editor to a different document type (e.g., Rate Limits), and then back to URL Maps, the incorrect value is no longer displayed.

If I switch to a different document type and then back to URL Maps, the deletion will not be saved and the old profile will appear :/

**Comment by spiros-p** [comment]
> If I switch to a different document type and then back to URL Maps, the deletion will not be saved and the old profile will appear :/

Yes, we're talking about slightly different scenarios. In my tests, I had forked the profile and immediately deleted it without ever saving it. 

I should have been more clear about this, and I've clarified my comment above. Sorry for the confusion.
